constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/Transition.java

```diff
@@ -274,26 +281,42 @@ float getDestinationPosition(float currentPosition, float velocity, float durati
                     return 1;
                 case ON_UP_STOP:
                     return Float.NaN;
-                case ON_UP_AUTOCOMPLETE:
                 case ON_UP_DECELERATE:
+                    return Math.max(0, Math.min(1, rest));
                 case ON_UP_DECELERATE_AND_COMPLETE:
+                   if (rest > 0.2f && rest < 0.8f){
+                       return rest;
+                } else {
+                       return rest > .5 ? 1 : 0;
```

Comment:
Not sure if this might cause a warning, but just in case, compare with another float: `rest > 0.5f`